services/agent/src/simcore_service_agent/modules/volumes_cleanup/_s3.py

```diff
@@ -60,9 +61,13 @@ def _get_s3_path(s3_bucket: str, labels: dict[str, str], volume_name: str) -> Pa
     return Path(f"/{joint_key}")
 
 
-async def _read_stream(stream):
+async def _read_stream(stream: StreamReader) -> str:
+    output = ""
     while line := await stream.readline():
-        logger.info(line.decode().strip("\n"))
+        message = line.decode()
+        output += message
+        logger.debug(message.strip("\n"))
```

Comment:
do you really need a log here ...  provided that you do log it as info later as well?
```python
    output = await _read_stream(process.stdout)
    await process.wait()
    logger.info(output)
```

Moreover, probably that log would "block" the advantage that you are seeking by async reading the string

Reply:
As replied above yes, to detect issues during development/debugging. If something is taking forever you cannot detect it.